### PR TITLE
csharp-ls: init at 0.8.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Add [csharp-ls](https://github.com/razzmatazz/csharp-language-server), a language server for C#.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 241185` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>csharp-ls</li>
  </ul>
</details>

File: pkgs/development/tools/language-servers/csharp-ls/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  nugetSha256 = "sha256-F3N6ESE/VmQA5sOMm3eqSkhSNVCCsnAXTCC+McsAnQw=";
+
+  dotnet-sdk = sdk_7_0;
+  dotnet-runtime = sdk_7_0;
+
+  meta = with lib; {
+    description = "Roslyn-based LSP language server for C#";
+    homepage = "https://github.com/razzmatazz/csharp-language-server";
+    changelog = "https://github.com/razzmatazz/csharp-language-server/releases/tag/v${version}";
```
Comment (wegank) [inline pkgs/development/tools/language-servers/csharp-ls/default.nix:21]: Does `version` here mean `lib.version`?

Comment (GaetanLepage) [reply pkgs/development/tools/language-servers/csharp-ls/default.nix:21]: No, it should be the package version. I have seen this way of specifying `meta.changelog` in various other packages such as: https://github.com/NixOS/nixpkgs/blob/3cee93dfaf3b5e9e9db719022fdb271c2f399aae/pkgs/games/freenukum/default.nix#L69

Comment (wegank) [reply pkgs/development/tools/language-servers/csharp-ls/default.nix:21]: Yeah, but that requires the `rec` keyword, right?

Comment (GaetanLepage) [reply pkgs/development/tools/language-servers/csharp-ls/default.nix:21]: Oh maybe ! I added it :)

**Review by wegank — APPROVED** [review]

File: pkgs/development/tools/language-servers/csharp-ls/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, buildDotnetGlobalTool
+, dotnetCorePackages
+}:
+let
+  inherit (dotnetCorePackages) sdk_7_0;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/language-servers/csharp-ls/default.nix:6]: This let in is not that useful. Either we write sdk_7_0 to a variable that stays static or we can inline it.

Comment (GaetanLepage) [reply pkgs/development/tools/language-servers/csharp-ls/default.nix:6]: Should I create a PR for this ? (I don't mind doing so)

Comment (SuperSandro2000) [reply pkgs/development/tools/language-servers/csharp-ls/default.nix:6]: You don't need to, just something I noticed which can be changed in the future.

**Pull request merged**